cpp/src/transform/jit/util/type.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Copyright 2018-2019 BlazingDB, Inc.
+ *     Copyright 2018 Christian Noboa Mardini <christian@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "type.h"
+#include <utilities/type_dispatcher.hpp>
+
+namespace cudf {
+namespace transformation {
+namespace jit {
+
+    /**---------------------------------------------------------------------------*
+     * @brief Functor to get type name in string
+     * 
+     * This functor uses the unofficial compiler macro __PRETTY_FUNCTION__
+     * to obtain the function signature which contains the template type name.
+     * The type name is searched and extracted from the string.
+     * 
+     * Example (clang): __PRETTY_FUNCTION__ =
+     * std::string type_name::operator()() [T = short]
+     * returns std::string("short")
+     * 
+     * Example (gcc): __PRETTY_FUNCTION__ =
+     * std::__cxx11::string type_name::operator()() [with T = short int; std::__cxx11::string = std::__cxx11::basic_string<char>]
+     * returns std::string("short int")
+     * 
+     * In case the type is wrapped using `wrapper`, the extra string "wrapper<" is
+     * also skipped to get the underlying wrapped type.
+     * 
+     *---------------------------------------------------------------------------**/
+    struct type_name {
```

Comment:
maybe move this and its counterpart in binary ops to Jit dir since this is common to all jit functionality we implement.